github-status/vendor/github.com/openfaas/openfaas-cloud/sdk/interfaces.go

```diff
@@ -18,3 +18,41 @@ func (l AuditLogger) Post(auditEvent AuditEvent) error {
 	PostAudit(auditEvent)
 	return nil
 }
+
+type StatusReporter interface {
+	Report(status *Status)
+}
+
+type NilStatusReporter struct {
+}
+
+func (r NilStatusReporter) Report(status *Status) {
+	return
+}
+
+type GithubStatusReporter struct {
+	HmacKey string
+	Gateway string
+}
+
+func (r GithubStatusReporter) Report(status *Status) {
+	ReportStatus(status, r.Gateway, r.HmacKey)
+}
+
+type Secret interface {
+	Read(string) (string, error)
+}
+
+type NilSecretReader struct {
+}
+
+func (r NilSecretReader) Read(key string) (string, error) {
+	return "", nil
+}
+
+type SecretReader struct {
+}
+
+func (r SecretReader) Read(key string) (string, error) {
+	return ReadSecret(key)
+}
```

Comment:
I'm not entirely sure if we should put `GithubStatusReporter` here? Maybe it would be good to move logic inside status.go file? and then instead of calling `ReportStatus` put the logic inside a method `Report`? I know that Alex did the same thing with `Logger` and `PostAudit` function, but I think that maybe we should have clean file with interfaces and maybe `nil` implementations? What do you think about that @alexellis